android-components/components/browser/state/src/main/java/mozilla/components/browser/state/action/AwesomeBarAction.kt

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.action
+
+import mozilla.components.concept.awesomebar.AwesomeBar
+
+/**
+ * [BrowserAction]s related to interactions with the [AwesomeBar].
+ */
+sealed class AwesomeBarAction : BrowserAction() {
+    /**
+     * Indicates that the suggestions displayed in the [AwesomeBar] have changed.
+     */
+    data class VisibilityStateUpdated(val visibilityState: AwesomeBar.VisibilityState) : AwesomeBarAction()
+
+    /**
+     * Indicates that the user clicked a [suggestion] in the [AwesomeBar].
+     */
+    data class SuggestionClicked(val suggestion: AwesomeBar.Suggestion) : AwesomeBarAction()
```

Comment:
My only light concern here is that the name might suggest a workflow that results in a suggestion being loaded into the browser. I don't have a good suggestion for a way to handle anything differently, though, so I'm happy to land as is.

Reply:
That makes sense; thanks for raising that! Maybe something like `SetClickedSuggestion` (and `SetVisibilityState` for the other action) might be a better name?

I liked the symmetry between `onVisibilityStateUpdated` / `AwesomeBarAction.VisibilityStateUpdated`, and `onSuggestionClicked` / `AwesomeBarAction.SuggestionClicked`, but I definitely see how it could imply additional semantics that aren't accurate.